### DISCUSSION TITLE
[Idea] Prestashop must make clear that it supports Windows

### DISCUSSION THREAD

**Discussion by musicpanda** [PrestaShop/PrestaShop#34821] [Ideas]
### The problem

Prestashop's policy towards Windows has always been mixed. It neither has totally supported it but neither has it totally rejected it.

Prestashop's main market is small shops. Most of those use shared hosting with some hosting provider. Some use a VPS. Own servers are rare. I never heard of someone using Windows as a live server. 

On the other hand Prestashop has always encouraged people to have a copy of their shop that they use for experimentation and development. And that copy is often run under Windows.

So while Prestashop never bothered to look at for example caching under Windows it didn't do anything that hindered running under Windows either. Until recently.

Originally Prestashop was wysiwyg. Even people who didn't know anything about software often managed to understand enough to make simple changes and fixes in their shop. As we see with Thirty Bees that path could have been followed further. But instead software "engineers" took over and decided to use a "professional" tool like Symfony. 

After many years in which most energy was wasted on having Prestashop fit into Symfony Prestashop was taken over and the organization was split. Unfortunately the "engineers" have taken over. They operate by the book and the book says that Prestashop doesn't support Windows. If they had looked better they would have seen that it doesn't support Windows as live server but that it encourages people to have a copy of their shop and that it is no problem if that is Windows.

The result is what can be considered as a real hate campaign against anything related to Windows. Suddenly the rule was introduced that the server should support SSL. Packages like Xampp and Wamp don't set this up correctly during installation and thus lots of people suddenly are confronted with a cryptic error message. Never mind that when you download such packages you are explicitly warned not to use them for production. 

For many people the first encounter with Prestashop is that - while looking for a webshop - they download it and run it under Windows to see how it looks and works. You can be sure that many of them won't go any further after encountering this error. As long as the software was maintained by a commercial organization that was understood. But now it seems that rigid formalism has taken over.

There will always be a diversity of servers. You have the different tastes of Linux. You have other Unix versions. You have IIS. You have the differences between Apache and Nginx. You have a variety of caching software. Just like with Windows all this variation is most of the time not a problem. But occasionally you need to take it into account. 

Recently I saw an error report on Github discarded just because the person in question saw some sign of Windows in the file paths. It wasn't clear at all that the error was Windows related. I have seen other reports been discarded as soon as it became clear that they were Windows specific. 

This hostility towards Windows rests on a misunderstanding on what Prestashop is about. Windows has always played an important part in Prestashop. When it is no longer compatible with Windows Prestashop will gradually lose a large part of its users. 

For that reason I think it should be explicitly established that this compatibility should be maintained.

### The solution

See above

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
It is unfortunate that you wrote this message yesterday because you were given a [2-month ban](https://github.com/PrestaShop/PrestaShop/issues/34799#issuecomment-1853696048) this morning due to repeated breaches of the [Code of Conduct](https://github.com/PrestaShop/PrestaShop/blob/ae98a9b763aff03a0d004332865a0f87a17ca556/CODE_OF_CONDUCT.md) . So you will not be able to answer this discussion for the next 2 months.
I will however provide a few answers.

# About PrestaShop direction

It seems you disagree with the path that the PrestaShop project is following. You're against Symfony, you're against SSL, you're against how we process issues.

I understand you preferred how things were in the time of PrestaShop 1.6, and Thirty Bees. I am sorry for you but your path and PrestaShop path seem to have diverged and we are not turning back. And there is nothing you can do about it: this is an open source project, we owe you nothing, you owe us nothing.

If you disagree with how this project is going there is a simple solution, thanks to the open source concept: you fork this repository and start building a fork yourself. You can build and maintain MusicPanda-PrestaShop and give it the direction you want ; this is how open source works. "If you don't like it, fork it". You are the captain of your own ship.

# About Windows

> Prestashop's main market is small shops

> they download it and run it under Windows to see how it looks and works

> When it is no longer compatible with Windows Prestashop will gradually lose a large part of its users.

Do you have evidence? Do you have statistics? Do you have data to support your claim?

If you have no evidence then you are saying opinions, not facts.

I'm afraid it's hard to listen to someone opinion when it's about changing radically the current direction of the software, especially since [you do not contribute to it](https://github.com/pulls?q=is%3Apr+author%3Amusicpanda+org%3Aprestashop). I listen more to people who invest in this software and collaborate to make it better.

On my side I have evidence and data (and I will soon present some in a blog post) that shows the project is very lively and in year 2023 we have achieved an all-time record in term of community growth. So I'm rather reassured that we're going the right way.

**Comment by kpodemski** [comment]
> After many years in which most energy was wasted on having Prestashop fit into Symfony Prestashop was taken over and the organization was split.

Ouch... I see that you really don't understand what these changes are for, what they are aimed at, and why it's so important to move in this direction. So, let me point out a few things :-)

First of all, PrestaShop is no longer a tool for everyone. The entry level into PrestaShop has increased significantly. That's a good thing. Why? For many years, PrestaShop was compared to Woocommerce, OpenCart and other solutions, which were dismissed after looking into the code by more experienced developers. PrestaShop was of interest to smaller players precisely because of the low entry level, there was no Webpack, or Symfony, and you didn't have to worry about server configuration or know JavaScript, TypeScript, Sass, or PHP better. Someone new to PrestaShop could download a PrestaShop release and start working. This has led to a situation where we still have a huge number of products in the ecosystem with very poor quality, they work in the beginning, they meet the business objectives, and then over time, they get worse and worse, and they are hard to maintain, and as merchants and their businesses grow they see that PrestaShop is no longer for them... 

But okay, we touched on the topic of merchant development, how their stores are growing, and merchants having higher expectations and requirements. You wrote that PrestaShop is for smaller stores and... I don't know if I agree. I don't know if currently, PrestaShop is a good solution for smaller stores. If they are conscious merchants who know what kind of investment it entails - then maybe yes. If they think the whole thing will go great, like investing 500 EUR in a server and a template, perhaps this is not the best strategy. Whether PrestaShop should target large or small merchants is neither your decision nor mine, only the project's main sponsor, the PrestaShop company. It would be very difficult to make it work for both.

Let's assume that you are right, let's assume that PrestaShop is mainly small stores. Should I assume that a small store = doing everything myself? Rather not. Because of this, more specialists are needed, PHP programmers, and front-end developers. And here we come to the point: the industry is becoming more and more difficult for developers. If one has to choose between learning a custom PrestaShop framework or Symfony, Symfony is probably the better option because it gives much more job security. This is one of the reasons why PrestaShop decided to rely on Symfony. We are opening up to the market of developers who do not want to be "PrestaShop developers", but simply "developers". Even if someone hasn't worked with PrestaShop, but is familiar with Symfony, he or she will be able to start with a project faster than if we continue on a codebase that was created more than 15 years ago and takes months to understand properly... By investing in Symfony, we can also enjoy the benefits of the PHP ecosystem. Creating solutions will ultimately become easier, faster, there will be more opportunities for software development. That's it regarding Symfony. We could still discuss this for hours, your ignorance surprised me.

Windows? Let me agree with you when it comes to the fact that many merchants who want to develop their own stores make a copy and try to put it on Windows, using Xampp, Laragon or other tools. I'll say it again: if you can't handle setting up a development environment on Windows, perhaps this solution is not for you. Maybe it would be better to contact a specialist. This may be a controversial opinion, but this is the reality. Btw. Often, the errors that PrestaShop faces are not directly related to how PrestaShop does some things, but its dependencies. 

Like @matks wrote, feel free to contribute to the project if you want it to work better under Windows. We'll focus on making PrestaShop a better tool for merchants who want to focus on their business and developers who know how to work on Windows, how to properly configure the environment, etc. 👍🏻

**Comment by MusicButterfly** [comment]
I am banned and I respect that. However, I see so many misunderstandings and falsehoods that I feel the need to at least once correct a few things. Don't worry, I won't do this again.
 - Florian and Matt accuse me of being a freeloader who doesn't contribute anything. In fact I am - under the name Musicmaster - one of the top contributers on the forum. I also brought some important problems into focus so that they could be fixed - including the fact that for a considerable time the Basic Edition couldn't be installed on a PHP version lower than 8.1 due to the way the ps_edition_basic module had been compiled. But I don't think this matters. Discussions should be had on arguments and their merits, not on the merits of those who say them. The fact that they suggest otherwise is in my eyes problematic.
 - I am the ultimate debugger and functionalist. Learning all the weird intricacies of Symfony or Github is for me almost the definition of wasting my time. Over time I may learn, but this is not where my heart is. On the other hand I have a quite detailed knowledge of the PS database structure.
 - So Github is for my taste a bit too complicated. It doesn't help that the few times I made an effort those seemed to disappear into the digital void. Probably even more important is that I find the Prestashop Github issues page a bit creepy place. Error reports are discarded by would-be dictators with flimsy arguments. And if you protest you are either ignored or put away as a troublemaker. Problems are seldom solved. Even when you provide complete instructions how the code could be fixed usually nothing happens. My main reason to still post problems there is documentation.
 - Prestashop forces people to make their Xampp/WAMP/Laragon installation SSL compatible - knowing very that they will never use that functionality. I see this is as a type of harrassment. I see the questions on the forum. And I know that there would be many more If I hadn't bothered to write detailed instructions on how to fix the problem.
 - I challenge the Prestashop staff to go to the forum and to answer people with those Windows installation problems directly. They could write something like: "We added this requirement because we hate Windows. The fact that wasted so much time on it is your own fault. It shows you are incompetent and we don't want incompetent users for Prestashop." That is the attitude you radiate anyway.
 - First impressions count. And the first impressions on the forum center on how many posts concern bugs and how many concern "how can I do this" questions. And in this context Windows errors definitely count as bugs.
 - Quality control for the Addons modules has been a problem for many years. Prestashop had a strict automatic script but didn't bother to check modules for bugs or basic functionality. Great to hear that they have improved quality by making it harder to make a module. Good social engineering!
 - Anyway, I opened this discussion post not about Symfony. It was about Windows and my request to officially support Windows so that people no longer have the excuse to close issues just because they happened under Windows. Neither of the reactions addresses that point.

**Comment by fabienvallon** [comment]
My name is Fabien, not Florian

Windows is not  officially supported ( and you can send patchs or create your own company to support it )

We can close.


PS: "  Error reports are discarded by would-be dictators with flimsy arguments. "' 
Ask @musicpanda to explain you the code of conduct.
He knows it by heart ;-)

**Comment by matthieu-rolland** [comment]
hello @MusicButterfly ,

Microsoft has acknowledged that Windows is not the optimal platform for web development, as the standard practice is to perform web development on a Unix system. This is one of the reasons why they offer solutions like the Windows Subsystem for Linux (WSL).

So now on Windows you can install a [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/install), it's completely integrated and offers good performance. 

You can keep working on your favorite OS, using your favorite IDE, and still have your PrestaShop running on a Linux system. 👍 

**EDIT:** If you're not familiar with unix/linux at all, it's ok, there are many tutorials on how to install a LAMP stack on a windows WSL: like [this one](https://www.it-connect.fr/installer-un-serveur-web-lamp-apache-mysql-php-sur-wsl-2/) . You can take it as an opportunity to learn new things, and sooner than you realize, you'll get to easily use tools that were developped with unix in mind and are a PITA to use on windows...
